src/redux/reducer.ts

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { createReducer } from '@reduxjs/toolkit';
+
+import {
+    getLocalStorageComputedLanguage,
+    getLocalStorageLanguage,
+    getLocalStorageTheme,
+    saveLocalStorageLanguage,
+    saveLocalStorageTheme,
+} from './local-storage';
+
+import {
+    ACTIVE_DIRECTORY,
+    ActiveDirectoryAction,
+    ADD_UPLOADING_ELEMENT,
+    AddUploadingElementAction,
+    AppActions,
+    ComputedLanguageAction,
+    CURRENT_CHILDREN,
+    CURRENT_PATH,
+    CurrentChildrenAction,
+    CurrentPathAction,
+    DIRECTORY_UPDATED,
+    DirectoryUpdatedAction,
+    LanguageAction,
+    REMOVE_UPLOADING_ELEMENT,
+    RemoveUploadingElementAction,
+    SEARCHED_ELEMENT,
+    SearchedElementAction,
+    SELECT_COMPUTED_LANGUAGE,
+    SELECT_DIRECTORY,
+    SELECT_LANGUAGE,
+    SELECT_THEME,
+    SelectDirectoryAction,
+    SELECTION_FOR_COPY,
+    SelectionForCopyAction,
+    SET_APPS_AND_URLS,
+    SetAppsAndUrlsAction,
+    ThemeAction,
+    TREE_DATA,
+    TreeDataAction,
+} from './actions';
+
+import {
+    AuthenticationActions,
+    AuthenticationRouterErrorAction,
+    AuthenticationRouterErrorState,
+    CommonMetadata,
+    CommonStoreState,
+    ElementAttributes,
+    ElementType,
+    GsLang,
+    GsLangUser,
+    GsTheme,
+    LOGOUT_ERROR,
+    LogoutErrorAction,
+    RESET_AUTHENTICATION_ROUTER_ERROR,
+    SHOW_AUTH_INFO_LOGIN,
+    ShowAuthenticationRouterLoginAction,
+    SIGNIN_CALLBACK_ERROR,
+    SignInCallbackErrorAction,
+    UNAUTHORIZED_USER_INFO,
+    UnauthorizedUserAction,
+    USER,
+    USER_VALIDATION_ERROR,
+    UserAction,
+    UserValidationErrorAction,
+} from '@gridsuite/commons-ui';
+import { PARAM_LANGUAGE, PARAM_THEME } from '../utils/config-params';
+import { UUID } from 'crypto';
+
+// IDirectory is exactly an IElement, with a specific type value
+export type IDirectory = ElementAttributes & {
+    type: ElementType.DIRECTORY;
+};
+
+export interface ElementAttributesES {
+    id: UUID;
+    name: string;
+    parentId: UUID;
+    type: ElementType;
+    owner: string;
+    subdirectoriesCount: number;
+    lastModificationDate: string;
+    pathName: string[];
+    pathUuid: UUID[];
+}
+
+export interface ITreeData {
+    rootDirectories: IDirectory[];
+    mapData: Record<string, IDirectory>;
+    initialized: boolean;
```

Comment:
How come this addition ?

Reply:
the reducer was incomplete/partial, I was obligated to add missing types